kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -0,0 +1,448 @@
+                //step one, reroute documents enroute to the error account Fiscal officer
+                if (node.equals("AccountFullEdit") && 
+                        pcardDocument.getTargetAccountingLine(0).getAccountNumber().equals(getErrorAccountNumber())) {
+                    pcardDocument.getTargetAccountingLine(0).setChartOfAccountsCode(StringUtils.isNotEmpty(procurementCardHolderDetail.getChartOfAccountsCode()) ? procurementCardHolderDetail.getChartOfAccountsCode() : getErrorChartCode());
+                    pcardDocument.getTargetAccountingLine(0).setAccountNumber(StringUtils.isNotEmpty(procurementCardHolderDetail.getAccountNumber()) ? procurementCardHolderDetail.getAccountNumber() : getErrorAccountNumber());
+                    pcardDocument.getTargetAccountingLine(0).setFinancialObjectCode(StringUtils.isNotEmpty(procurementCardHolderDetail.getFinancialObjectCode()) ? procurementCardHolderDetail.getFinancialObjectCode() : getDefaultObjectCode());
+                    pcardDocument.getTargetAccountingLine(0).setSubAccountNumber(procurementCardHolderDetail.getSubAccountNumber());
+                    pcardDocument.getTargetAccountingLine(0).setFinancialSubObjectCode(procurementCardHolderDetail.getFinancialSubObjectCode());
+                    if ( LOG.isInfoEnabled() ) {
+                        LOG.info("Rerouting doc #" + pcardDocumentId + " at AccountFullEdit for error account Fiscal Officer.");
+                    }                        
+                    requeueDocument(pcardDocument, pcardDocumentId, node, "HasReconciler"); 
+                    // pause between reroutes
+                    try {
+                        Thread.sleep(3000);
+                    }
+                    catch (InterruptedException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+                
+              //step two, reroute documents enroute to the reconciler group(s) that had group membership problems
+                else if (node.equals("ProcurementCardReconciler") && groupRerouting && 
+						parameterService.getParameter(ProcurementCardRerouteDocumentsStep.class,
+								RECONCILER_GROUPS_TO_REROUTE_PARM_NM).getValue() != null
+						&& parameterService
+								.getParameter(ProcurementCardRerouteDocumentsStep.class,
+										RECONCILER_GROUPS_TO_REROUTE_PARM_NM)
+								.getValue().contains(procurementCardHolderDetail.getReconcilerGroupId())) {
+                    if ( LOG.isInfoEnabled() ) {
+						LOG.info("Rerouting doc #" + pcardDocumentId + " at ProcurementCardReconciler, in "
+								+ RECONCILER_GROUPS_TO_REROUTE_PARM_NM + " group "
+								+ procurementCardHolderDetail.getReconcilerGroupId());
+                    }
+                    requeueDocument(pcardDocument, pcardDocumentId, node, "ProcurementCardReconciler"); 
+                    // pause between reroutes
+                    try {
+                        Thread.sleep(3000);
```

Comment:
I'd buy that. I'd also say that any effort towards fixing these threading sleep hacks (or even just making them less terrible) should be on a separate ticket.